packages/server/src/middlewares/domainBlacklist.ts

```diff
@@ -0,0 +1,98 @@
+// middlewares/domainBlacklist.ts
+import type { Request, Response, NextFunction } from "express";
+import logger from "../utils/logger";
+import error from "../errorResponse.json";
+
+export function isValidDomain(domain: string) {
+  if (typeof domain !== "string") return false;
+
+  const trimmed = domain.trim().toLowerCase();
+  if (trimmed.length === 0 || trimmed.length > 255) return false;
+
+  const domainRegex =
+    /^(?!-)(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]$/;
+
+  return domainRegex.test(trimmed);
+}
+
+export const parseBlacklistedDomains = (rawDomains: string) => {
+  const result = new Set<string>();
+
+  if (!rawDomains || typeof rawDomains !== "string") return result;
+
+  rawDomains
+    .split(",")
+    .map((domain) => domain.trim().toLowerCase())
+    .forEach((domain) => {
+      if (!domain) return;
+      if (!isValidDomain(domain)) {
+        logger.warn(`Invalid domain in blacklist config: "${domain}"`);
+        return;
+      }
+      result.add(domain);
+    });
+
+  return result;
+};
+
+export class BlacklistManager {
+  private cachedBlacklist: Set<string> | null = null;
+
+  constructor(
+    private rawDomains: string = process.env.LOGCHIMP_BLACKLISTED_DOMAINS || "",
+  ) {}
+
+  getBlacklistedDomains() {
+    if (!this.cachedBlacklist) {
+      this.cachedBlacklist = parseBlacklistedDomains(this.rawDomains);
+    }
+    return this.cachedBlacklist;
```

Comment:
Can you reverse this check?

If there is a cache, return the cache, else compute the fresh-new value.